desc/objectives/_neoclassical.py

```diff
@@ -0,0 +1,253 @@
+        desired bound (lower or upper) of the profile at those locations.
+        Defaults to ``bounds=(0, np.inf)``
+    weight : {float, ndarray}, optional
+        Weighting to apply to the Objective, relative to other Objectives.
+        Must be broadcastable to Objective.dim_f
+    normalize : bool, optional
+        Whether to compute the error in physical units or non-dimensionalize.
+    normalize_target : bool, optional
+        Whether target and bounds should be normalized before comparing to computed
+        values. If `normalize` is `True` and the target is in physical units,
+        this should also be set to True.
+    loss_function : {None, 'mean', 'min', 'max'}, optional
+        Loss function to apply to the objective values once computed. This loss function
+        is called on the raw compute value, before any shifting, scaling, or
+        normalization.
+    deriv_mode : {"auto", "fwd", "rev"}
+        Specify how to compute Jacobian matrix, either forward mode or reverse mode AD.
+        "auto" selects forward or reverse mode based on the size of the input and output
+        of the objective. Has no effect on self.grad or self.hess which always use
+        reverse mode and forward over reverse mode respectively.
+    grid : Grid, optional
+        Collocation grid to evaluate flux surface averages.
+        Should have poloidal and toroidal resolution.
+    alpha : ndarray
+        Unique coordinate values for field line poloidal angle label alpha.
+    zeta : ndarray
+        Unique coordinate values for field line following coordinate zeta. Must be
+        strictly increasing. A good reference density is 100 knots per toroidal transit.
+        For axisymmetric devices only one toroidal transit is necessary. Otherwise,
+        more toroidal transits will give more accurate result, with diminishing returns.
+    num_quad : int
+        Resolution for quadrature of bounce integrals. Default is 31,
+        which gets sufficient convergence, so higher values are likely unnecessary.
+    num_pitch : int
+        Resolution for quadrature over velocity coordinate, preferably odd.
+        Default is 99. Effective ripple will look smoother at high values.
+        (If computed on many flux surfaces and micro oscillation is seen
+        between neighboring surfaces, increasing num_pitch will smooth the profile).
+    batch : bool
+        Whether to vectorize part of the computation. Default is true.
+    name : str, optional
+        Name of the objective function.
+
+    """
+
+    _coordinates = "r"
+    _units = "~"
+    _print_value_fmt = "Effective ripple ε¹ᐧ⁵: {:10.3e} "
+
+    def __init__(
+        self,
+        eq,
+        target=None,
+        bounds=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        loss_function=None,
+        deriv_mode="auto",
+        grid=None,
+        alpha=np.array([0]),
+        zeta=np.linspace(0, 2 * np.pi, 100),
```

Comment:
i think we should default to a higher number of transits if its usually necessary

Reply:
I increased it, but we can't match Neo's default num transits due to memory issues